llvm/lib/Target/Xtensa/XtensaISelLowering.cpp

```diff
@@ -338,16 +329,13 @@ SDValue XtensaTargetLowering::LowerFormalArguments(
     // Arguments stored on registers
     if (VA.isRegLoc()) {
       EVT RegVT = VA.getLocVT();
-      const TargetRegisterClass *RC;
 
-      if (RegVT == MVT::i32)
-        RC = &Xtensa::ARRegClass;
-      else
+      if (RegVT != MVT::i32)
         report_fatal_error("RegVT not supported by FormalArguments Lowering");
 
       // Transform the arguments stored on
       // physical registers into virtual ones
-      unsigned Register = MF.addLiveIn(VA.getLocReg(), RC);
+      unsigned Register = MF.addLiveIn(VA.getLocReg(), &Xtensa::ARRegClass);
```

Comment:
```suggestion
      Register Reg = MF.addLiveIn(VA.getLocReg(), &Xtensa::ARRegClass);
```
Use Register type and try to avoid shadowing type names with variable names

Reply:
Thank you very much for comments. Fixed.